Framework/PythonInterface/plugins/algorithms/WorkflowAlgorithms/LagrangeILLReduction.py

```diff
@@ -232,6 +289,16 @@ def get_water_correction(self, correction_file: str) -> np.ndarray:
             self.log().warning("Water correction file is faulty.")
             self.log().warning(str(e))
             correction = None
+        else:
+            # if the sample is in energy transfer unit, we need to make the correction consistent by subtracting constant offset
+            if not self.use_incident_energy:
+                correction[:, 0] -= self.INCIDENT_ENERGY_OFFSET
+            water_corr_ws = "__{}_waterCorrection".format(self.output_ws_name)
+            CreateWorkspace(OutputWorkspace=water_corr_ws,
+                            DataX=correction[:, 0],
+                            DataY=correction[:, 1],
+                            UnitX='Energy')
+            self.intermediate_workspaces.append(water_corr_ws)
         return correction
 
     def merge_adjacent_points(self, data: np.ndarray) -> np.ndarray:
```

Comment:
What happens to metadata (temperature and time) when merging adjacent points?

Reply:
That's a very good point. I changed this behaviour to mask the input `ndarray` so that the temperature and time is also masked for points that are not taken into account.